File: app/src/main/java/net/daverix/ajvm/operation/ISubOperation.java
package net.daverix.ajvm.operation;


import net.daverix.ajvm.Frame;
import net.daverix.ajvm.ByteCodeReader;

import java.io.IOException;

public class ISubOperation implements ByteCodeOperation {
    @Override
    public void execute(ByteCodeReader reader,
                        int indexOfBytecode,
                        Frame currentFrame) throws IOException {
        int iSub2 = (int) currentFrame.pop();
        int iSub1 = (int) currentFrame.pop();
        currentFrame.push(iSub1 - iSub2);
    }
}

File: app/src/main/java/net/daverix/ajvm/operation/ISubOperation.kt
package net.daverix.ajvm.operation


import net.daverix.ajvm.Frame
import net.daverix.ajvm.ByteCodeReader

import java.io.IOException

class ISubOperation : ByteCodeOperation {
    @Throws(IOException::class)
    override fun execute(reader: ByteCodeReader,
                         indexOfBytecode: Int,
                         currentFrame: Frame) {
        val iSub2 = currentFrame.pop() as Int
        val iSub1 = currentFrame.pop() as Int
        currentFrame.push(iSub1 - iSub2)
    }
}
